src/Ghcid.hs

```diff
@@ -192,29 +200,41 @@ mainWithTerminal termSize termOutput =
                 opts <- return $ opts{restart = nubOrd $ (origDir </> ".ghcid") : restart opts, reload = nubOrd $ reload opts}
                 when (topmost opts) terminalTopmost
 
-                termSize <- return $ case (width opts, height opts) of
-                    (Just w, Just h) -> return $ TermSize w h WrapHard
+                termSize <- case (width opts, height opts) of
+                    (Just w, Just h) -> return $ TermSize w (Just h) WrapHard
                     (w, h) -> do
                         term <- termSize
                         -- if we write to the final column of the window then it wraps automatically
                         -- so putStrLn width 'x' uses up two lines
                         return $ TermSize
                             (fromMaybe (pred $ termWidth term) w)
-                            (fromMaybe (termHeight term) h)
+                            (h <|> termHeight term)
                             (if isJust w then WrapHard else termWrap term)
 
+                termSize <- return $
+                  if no_height_limit opts
+                  then termSize { termHeight = Nothing }
+                  else termSize
+
+                supportsANSI <- hSupportsANSI stdout
+
                 restyle <- do
-                    useStyle <- case color opts of
```

Comment:
I'd rather not even query hSupportsANSI unless we have to, since it's the kind of thing I can imagine goes wrong in odd circumstances. Suggest using `once` to capture its value, and then use it in the two places (if needed). But see the remark below.